modules/compactor/compactor.go

```diff
@@ -211,9 +213,30 @@ func (c *Compactor) Owns(hash string) bool {
 	return rs.Instances[0].Addr == ringAddr
 }
 
-// Combine implements common.ObjectCombiner
-func (c *Compactor) Combine(dataEncoding string, objs ...[]byte) ([]byte, bool, error) {
-	return model.ObjectCombiner.Combine(dataEncoding, objs...)
+// Combine implements tempodb.CompactorSharder
+func (c *Compactor) Combine(dataEncoding string, tenantID string, objs ...[]byte) ([]byte, bool, error) {
+	combinedObj, wasCombined, err := model.ObjectCombiner.Combine(dataEncoding, objs...)
+	if err != nil {
+		return nil, false, err
+	}
+
+	maxBytes := c.overrides.MaxBytesPerTrace(tenantID)
+	if len(combinedObj) < maxBytes {
```

Comment:
**question:** Both the ingester and the [search PR](https://github.com/grafana/tempo/pull/1318) treat this as 0=optional.  Let's do the same here?  Else I think this logic will stop discarding everything.